src/developer-quickstart.md

```diff
@@ -0,0 +1,482 @@
+# Developer Quickstart
+
+This guide will walk developers through writing a smart contract in Sway, a simple test, deploying to Fuel, and building a frontend.
+
+Before we begin, it may be helpful to understand terminology that will used throughout the docs and how they relate to each other:
+
+- **Fuel**: the Fuel blockchain.
+- **FuelVM**: the virtual machine powering Fuel.
+- **Sway**: the domain-specific language crafted for the FuelVM; it is inspired by Rust.
+- **Forc**: the build system and package manager for Sway, similar to Cargo for Rust.
+
+## Understand Sway Program Types
+
+There are four types of Sway programs:
+
+- `contract`
+- `predicate`
+- `script`
+- `library`
+
+Contracts, predicates, and scripts can produce artifacts usable on the blockchain, while a library is simply a project designed for code reuse and is not directly deployable.
+
+The main features of a smart contract that differentiate it from scripts or predicates are that it is callable and stateful.
+
+A script is runnable bytecode on the chain which can call contracts to perform some task. It does not represent ownership of any resources and it cannot be called by a contract.
+
+|             | deployable on the blockchain:       | can have state:      | callable on the blockchain: | designed for code reuse: |
+|-------------|-------------------------------------|----------------------|-----------------------------|--------------------------|
+| contract  | ✅                                  | ✅                   | ✅                          | ❌                        |
+| predicate | ✅                                  | ❌                    | ❌                           | ❌                        |
+| script    | ❌                                   | ❌                    | ❌                           | ❌                        |
+| library   | ✅ (via a contract or predicate) | ❌ | ❌                           | ✅                       |
+
+See [the chapter on program types](../sway-program-types/index.md) for more information.
+
+## Your First Sway Project
+
+We'll build a simple counter contract with two functions: one to increment the counter, and one to return the value of the counter.
+
+A few pieces of info that will be helpful before moving on:
+
+- This guide was created using VSCode as the code editor.
+- Download the Sway language extension in VSCode to get syntax highlighting, keyboard shortcuts, and more.
+- Download the rust-analyzer extension in VSCode to get syntax highlighting, code completion, and more.
+
+**Start by creating a new, empty folder. We'll call it `fuel-project`.**
+
+### Writing the Contract
+
+First, let's [install the Sway toolchain](https://github.com/FuelLabs/fuelup).
+
+Then with `forc` installed, create a contract project inside of your `fuel-project` folder:
+
+```sh
+cd fuel-project
+forc new counter_contract
+```
+
+Here is the project that `Forc` has initialized:
+
+```console
+├── fuel-project
```

Comment:
`console` blocks should be used with the commands that produce the output, preceded by a `$`. E.g. (not sure this is the actual command)

````suggestion
```console
$ tree .
├── fuel-project
````

Reply:
this is just an illustration of the folder structure, I didn't use a command for this. But I will do this everywhere else